src/libraries/System.Diagnostics.Tracing/ref/System.Diagnostics.Tracing.Counters.cs

```diff
@@ -11,26 +11,31 @@ public abstract partial class DiagnosticCounter : System.IDisposable
         public System.Diagnostics.Tracing.EventSource EventSource { get { throw null; } }
         public string Name { get { throw null; } }
         public void AddMetadata(string key, string? value) { }
+        [System.Runtime.Versioning.UnsupportedOSPlatform("browser")]
```

Comment:
This is kind of odd IMO. If a class implements IDisposable, all callers should call Dispose on it. So if we are saying this class is supported on browser, but you shouldn't dispose of it, it is going to make for some odd coding patterns. For example, you can't use `using`. Why not mark the whole class as unsupported?

Reply:
This is because this is disposing `CounterGroup` inside, `CounterGroup.Dispose()` is not calling any platform-specific code, just the entire type annotated with `UnsupportedOSPlatform("browser")`, i was thinking to suppress the warning on `CounterGroup.Dispose()`, but looking again: it is an abstract class and all derived types calling (and supposed to call ) the its only platform-specific API https://github.com/dotnet/runtime/blob/f6d961850f4b943a52461efd7ed4e2ad8e7a3363/src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/DiagnosticCounter.cs#L52-L53. So i think its better to annotate entire class